features/ticketing/daemon/src/main/java/org/opennms/netmgt/ticketd/TroubleTicketer.java

```diff
@@ -150,17 +151,27 @@ public String getName() {
     @Override
 	public void onEvent(Event e) {
         try {
-		if (EventConstants.TROUBLETICKET_CANCEL_UEI.equals(e.getUei())) {
-			handleCancelTicket(e);
-		} else if (EventConstants.TROUBLETICKET_CLOSE_UEI.equals(e.getUei())) {
-			handleCloseTicket(e);
-		} else if (EventConstants.TROUBLETICKET_CREATE_UEI.equals(e.getUei())) {
-			handleCreateTicket(e);
-		} else if (EventConstants.TROUBLETICKET_UPDATE_UEI.equals(e.getUei())) {
-			handleUpdateTicket(e);
-		} else if (isReloadConfigEvent(e)) {
-            handleTicketerReload(e);
- 		}
+            if (EventConstants.TROUBLETICKET_CANCEL_UEI.equals(e.getUei())) {
+                handleCancelTicket(e);
+            } else if (EventConstants.TROUBLETICKET_CLOSE_UEI.equals(e.getUei())) {
+                handleCloseTicket(e);
+            } else if (EventConstants.TROUBLETICKET_CREATE_UEI.equals(e.getUei())) {
+                handleCreateTicket(e);
+            } else if (EventConstants.TROUBLETICKET_UPDATE_UEI.equals(e.getUei())) {
+                handleUpdateTicket(e);
+            } else if (isReloadConfigEvent(e)) {
+                Parm parm = e.getParm(EventConstants.PARM_DAEMON_NAME);
+                if ("Ticketd".equalsIgnoreCase(parm.getValue().getContent())) {
+                    DaemonTools.handleReloadEvent(e, "ticketd", event -> {
+                        handleTicketerReload(event);
+                    });
+                }
+                if ("Ticketer".equalsIgnoreCase(parm.getValue().getContent())) {
+                    DaemonTools.handleReloadEvent(e, "ticketer", event -> {
+                        handleTicketerReload(event);
+                    });
+                }
+            }
```

Comment:
This can be written more elegantly, by first determining the daemon name and afterwards invoking `DaemonTools`.
However in NMS-10350 the daemon seems to be not reloadable at all, so why bother: https://github.com/OpenNMS/opennms/pull/2160/files